pop/service/service_no_race_test.go

```diff
@@ -0,0 +1,81 @@
+// +build !race
+
+package service
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"gopkg.in/dedis/onet.v1"
+	"gopkg.in/dedis/onet.v1/crypto"
+	"gopkg.in/dedis/onet.v1/log"
+	"gopkg.in/dedis/onet.v1/network"
+
+	"fmt"
+)
+
+func TestService_MergeConfig(t *testing.T) {
+	local := onet.NewTCPTest()
+	defer local.CloseAll()
+	nbrNodes := 4
+	nbrAtt := 4
+	nodes, r, _ := local.GenTree(nbrNodes, true)
+
+	descs, atts, srvcs, priv := storeDescMerge(local.GetServices(nodes, serviceID), r, nbrAtt)
+	hash := make([]string, nbrNodes/2)
+	hash[0] = string(descs[0].Hash())
+	hash[1] = string(descs[1].Hash())
+	cc := &mergeConfig{srvcs[0].data.Finals[hash[0]], []byte{}}
+	srvcs[0].SendRaw(r.List[1], cc)
+	mcr := <-srvcs[0].syncs[hash[0]].mcChannel
+	require.NotNil(t, mcr)
+	require.Nil(t, mcr.Final)
+	require.Equal(t, PopStatusWrongHash, mcr.PopStatus)
+
+	require.Equal(t, nbrAtt, len(atts))
+
+	cc.ID = []byte(hash[1])
+	srvcs[0].SendRaw(r.List[2], cc)
+	mcr = <-srvcs[0].syncs[hash[0]].mcChannel
+	require.NotNil(t, mcr)
+	require.Nil(t, mcr.Final)
+	require.Equal(t, PopStatusMergeNonFinalized, mcr.PopStatus)
+	// finish parties
+	for i, desc := range descs {
+		descHash := desc.Hash()
+
+		fr := &finalizeRequest{}
+		fr.DescID = descHash
+		fr.Attendees = atts[2*i : 2*i+2]
+		hash, err := fr.hash()
+		sg, err := crypto.SignSchnorr(network.Suite, priv[2*i], hash)
+		log.ErrFatal(err)
+		fr.Signature = sg
+		_, err = srvcs[2*i].FinalizeRequest(fr)
+		require.NotNil(t, err)
+
+		sg, err = crypto.SignSchnorr(network.Suite, priv[2*i+1], hash)
+		log.ErrFatal(err)
+		fr.Signature = sg
+		msg, err := srvcs[2*i+1].FinalizeRequest(fr)
+		require.Nil(t, err)
+		require.NotNil(t, msg)
+		_, ok := msg.(*finalizeResponse)
+		require.True(t, ok)
+	}
+
+	log.Info("Group 1, Server:", srvcs[0].ServerIdentity())
+	log.Info("Group 1, Server:", srvcs[1].ServerIdentity())
+	log.Info("Group 2, Server:", srvcs[2].ServerIdentity())
+	log.Info("Group 2, Server:", srvcs[3].ServerIdentity())
+	cc.Final = srvcs[0].data.Finals[hash[0]]
+	cc.ID = []byte(hash[1])
+	srvcs[0].SendRaw(r.List[2], cc)
+	meta := srvcs[2].data.merges[hash[1]]
+	// Here is involuntary race condition solved by waiting in cycle
+	// on timeout
+	// In this case I can't wait till the end of process because
+	// I test here only one message
+	Eventually(t, func() bool { return len(meta.statementsMap) == len(descs) },
```

Comment:
If the `Eventually` creates races, then it should go away. Either you wait for the message, or you create a channel that you can wait on.